integration-tests/micrometer-prometheus/src/main/java/io/quarkus/it/micrometer/prometheus/PathTemplateResource.java

```diff
@@ -4,10 +4,13 @@
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.PathParam;
 
+import io.smallrye.common.vertx.ContextLocals;
+
 @Path("template/path/{value}")
 public class PathTemplateResource {
     @GET
     public String get(@PathParam("value") String value) {
+        ContextLocals.put("context-local", "val-" + value);
```

Comment:
This is a bad practice. We should not encourage ppl to add tags from path params. Even if they think the nr of different values are limited, they cannot really control it.

Reply:
It might be bad practice,  but people shouldn't be limited by accidental technically reasons from doing it.

ThreadLocals should not be used generally,  but that doesn't mean we should prevent their use.

Reply:
Fair, but this should come with a big warning that they will shoot themselves in the foot if they don't consider cardinatility.
See for exampleL [#users > Are you using path parameters? (metrics)](https://quarkusio.zulipchat.com/#narrow/channel/187030-users/topic/Are.20you.20using.20path.20parameters.3F.20.28metrics.29/with/518040545)

I think a warning should be placed in the docs and a in the test.

Reply:
Because in the end, ppl will complain and we will spend time debugging this stuff.

Reply:
I would add we should also place a warning in the Javadoc

Reply:
If you want to add something in the various places, I'll gladly approve

Reply:
For context: my initial use case is not related to path params, it's related to logging the tenant ID associated with the request. And tenant IDs are guaranteed to form a finite set, although they could be like a thousand or so. Are you saying that even this use case is bad practice?

Reply:
Yes. It must not be used like that. It will blow the dimensions exponentially. Remember that for each tenant id you will have different metrics per endpoint, response status and method.
Place that data in the span and later query the APM's span attributes and correlate with exemplars.